src/storage/mapping/BaseFileIdentifierMapper.ts

```diff
@@ -41,12 +40,6 @@ export class BaseFileIdentifierMapper implements FileIdentifierMapper {
    */
   public async mapUrlToFilePath(identifier: ResourceIdentifier, isMetadata: boolean, contentType?: string):
   Promise<ResourceLink> {
-    // Technically we could allow paths ending on .meta as long as we make sure there is never a mixup.
-    // But this can lead to potential issues.
-    // This also immediately stops users that expect they can update metadata like this.
-    if (this.isMetadataPath(identifier.path)) {
-      throw new ConflictHttpError('Not allowed to create files with the metadata extension.');
-    }
```

Comment:
This made me think if it is even necessary for this class to be aware of the fact that there is something like `metadata`, but unfortunately we need metadata to set the extension when writing to a file and I can't immediately think of a way around this.